### ISSUE TITLE
Server will not find/list packages in packages directory.

### ISSUE THREAD

**Issue by mvarga6** [TanukiSharp/MinimalNugetServer#13]
I've configured the server with 

```
{
  "server": {
    "url": "http://*:4356"
  },
  "nuget": {
    "packages": "/home/mike/.nuget_local",
    "makeReadonly": true
  },
  "cache": {
    "type": "NoCacheLoadAll",
    "duration": -1
  }
}
```

which claims to read the packages on startup,

```
Cache strategy: NoCacheLoadAll
Processing packages...
Processing packages done
```

and I have added a single nupkg file to source directory,

```
>>> ls /home/mike/.nuget_local/
MyTestPackage.1.0.0.nupkg
```

and added the feed to nuget

```
Registered Sources:

  1.  https://www.nuget.org/api/v2/ [Enabled]
      https://www.nuget.org/api/v2/
  2.  localv2 [Enabled]
      http://localhost:4356/v2/
```

but I can't get the server to actually _serve_ this file.

```
>>> nuget list -Source localv2
No packages found.
```

Server log for that request:
```
---
ContentLength:
ContentType:
Method: GET
Path: /v2/
PathBase:
Protocol: HTTP/1.0
QueryString:
---


---
ContentLength:
ContentType:
Method: GET
Path: /v2/$metadata
PathBase:
Protocol: HTTP/1.0
QueryString:
---


---
ContentLength:
ContentType:
Method: GET
Path: /v2/Packages
PathBase:
Protocol: HTTP/1.1
QueryString:
---
```

*What am I doing wrong?* Seems like I've followed everything to a T, and this just isn't working as expected.  OR I'm completely missing something. @TanukiSharp any idea why this is happening?

Cheers

**Comment by TanukiSharp** [comment]
Your configuration looks correct.
Could you `ls -la` in `/home/mike/` and possibly in `/home/mike/.nuget_local` just to make sure of the access rights ?

Maybe a bug with the `.` symbol, can you try to rename `.nuget_local` to `nuget_local` just to see if it changes something ?

If that doesn't change anything, I will try with the same path and see how it behaves.

**Comment by mvarga6** [comment]
Thanks for the quick response!

The access rights to that directory are
```
drwxr-xr-x   3 mike mike  4096 Nov 25 20:14 .nuget_local
```

### TL;DR
Changing `.nuget_local` to `nuget_local` completely solves the issue. 

### Work around keeping `.nuget_local` naming

As for a work around, if you add some structured/extracted NuGet packages to the same directory, I was able to get it to serve them, i.e. manually add the nupkg file to that directory using nuget, 
```
nuget add MyTestPackage.1.0.0.nupkg -s ~./nuget_local
```
so `.nuget_local` has this structure:
```
mike:~$ tree .nuget_local
.nuget_local
└── mytestpackage
    ├── 1.0.0
    │   ├── mytestpackage.1.0.0.nupkg
    │   ├── mytestpackage.1.0.0.nupkg.sha512
    │   └── mytestpackage.nuspec
    ├── 1.0.1
        ├── mytestpackage.1.0.1.nupkg
        ├── mytestpackage.1.0.1.nupkg.sha512
        └── mytestpackage.nuspec
```
and then resulting list from that source is now
```
mike:~$ nuget list -Source localv2
mytestpackage 1.0.0
mytestpackage 1.0.1
```

So it seems there's no issue if the packages are in the structured format but if I just drop `.nupkg` files into that dir directly, it doesn't find them _when_ there is `.` in the directory name. 

I'd try it out for yourself and see if you can reproduce it.  If not, its possible there was a config thing that changed/fixed itself when I rebooted.  ¯\_(ツ)_/

**Comment by TanukiSharp** [comment]
Could you tell me your version of `nuget` please ?
I have NuGet version 3.5.0.1938 here and I get different results, but not satisfying ones...

Microsft changed the format of NuGet protocol a lot and I can't keep up with the pace, and I don't want to implement support for all the version.

I wish Microsoft could make a NuGet server open-source that runs on .NET Core.
